### PR TITLE
Clean up the code

### PR DESCRIPTION
**Pull request by kunichik**
Clean up: place classes and their implementations in corresponding files

### CODE REVIEW

File: src/ipc/dbus/DBusIPCProxyBinder.h [hunk 1]
```diff
@@ -61,10 +61,7 @@ class FaceliftIPCLibDBus_EXPORT DBusIPCProxyBinder : public IPCProxyBinderBase
 
     DBusIPCProxyBinder(InterfaceBase &owner, QObject *parent = nullptr);
 
-    const QString &interfaceName() const
-    {
-        return m_interfaceName;
-    }
+    const QString &interfaceName() const;
```
Comment (bitmouse) [inline src/ipc/dbus/DBusIPCProxyBinder.h:64]: inline functions should stay in headers

File: src/ipc/dbus/DBusIPCProxyBinder.h [hunk 2]
```diff
@@ -80,10 +77,7 @@ class FaceliftIPCLibDBus_EXPORT DBusIPCProxyBinder : public IPCProxyBinderBase
 
     void setServiceAvailable(bool isRegistered);
 
-    bool isServiceAvailable() const override
-    {
-        return m_serviceAvailable;
-    }
+    bool isServiceAvailable() const override;
```
Comment (bitmouse) [inline src/ipc/dbus/DBusIPCProxyBinder.h:80]: as above

File: src/ipc/dbus/DBusIPCProxyBinder.h (hunk 1 above)
Comment (bitmouse) [reply src/ipc/dbus/DBusIPCProxyBinder.h:64]: This applies to other cases too

File: src/ipc/ipc-common/IPCProxyBaseBase.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+class FaceliftIPCCommonLib_EXPORT IPCProxyBaseBase {
+
+public:
+
+    void deserializeCommonSignal(facelift::CommonSignalID signalID, InterfaceBase* i);
+
+};
+
+}
```
Comment (bitmouse) [inline src/ipc/ipc-common/IPCProxyBaseBase.h:51]: Add new line at the end of file!

File: src/ipc/ipc-common/LocalProviderBinderBase.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+void LocalProviderBinderBase::init()
+{
+    m_interfaceManager.content().addListener(m_proxy.objectPath(),
+            this, &LocalProviderBinderBase::checkLocalAdapterAvailability);
+    checkLocalAdapterAvailability();
+}
+
+}
```
Comment (bitmouse) [inline src/ipc/ipc-common/LocalProviderBinderBase.cpp:53]: new line

File: src/ipc/ipc-common/LocalProviderBinderBase.h [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    virtual void checkLocalAdapterAvailability() = 0;
+
+    void init();
+
+protected:
+    IPCProxyNewBase &m_proxy;
+    InterfaceManager &m_interfaceManager = InterfaceManager::instance();
+};
+
+}
```
Comment (bitmouse) [inline src/ipc/ipc-common/LocalProviderBinderBase.h:59]: new line

File: src/ipc/ipc-common/NotAvailableImplBase.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+class FaceliftIPCCommonLib_EXPORT NotAvailableImplBase
+{
+
+protected:
+    static void logMethodCall(const InterfaceBase &i, const char *methodName);
+    static void logSetterCall(const InterfaceBase &i, const char *propertyName);
+    static void logGetterCall(const InterfaceBase &i, const char *propertyName);
+};
+
+}
```
Comment (bitmouse) [inline src/ipc/ipc-common/NotAvailableImplBase.h:51]: new line

File: src/ipc/ipc-common/OutputPayLoad.cpp [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+OutputPayLoad::OutputPayLoad(QByteArray &payloadArray) : m_payloadArray(payloadArray), m_dataStream(&m_payloadArray, QIODevice::WriteOnly)
+{
+}
+
+const QByteArray &OutputPayLoad::getContent() const
+{
+    return m_payloadArray;
+}
+
+}
```
Comment (bitmouse) [inline src/ipc/ipc-common/OutputPayLoad.cpp:43]: new line

File: src/ipc/local/CMakeLists.txt [hunk 8]
```diff
@@ -1,14 +1,23 @@
...
         LocalIPCProxy.h
         LocalIPCRegistry.h
         LocalIPC-serialization.h
         LocalIPCServiceAdapter.h
+        FaceliftIPCCommon.h
     LINK_LIBRARIES FaceliftIPCCommonLib FaceliftModelLib
     PUBLIC_HEADER_BASE_PATH ${CMAKE_CURRENT_SOURCE_DIR}
     MONOLITHIC_SUPPORTED
-)
+)
```
Comment (bitmouse) [inline src/ipc/local/CMakeLists.txt:23]: new line

File: src/ipc/ipc-common/LocalProviderBinderBase.h (hunk 5 above)
Comment (kunichik) [reply src/ipc/ipc-common/LocalProviderBinderBase.h]: done

File: src/ipc/ipc-common/NotAvailableImplBase.h (hunk 6 above)
Comment (kunichik) [reply src/ipc/ipc-common/NotAvailableImplBase.h]: done

File: src/ipc/ipc-common/OutputPayLoad.cpp (hunk 7 above)
Comment (kunichik) [reply src/ipc/ipc-common/OutputPayLoad.cpp]: done

File: src/ipc/local/CMakeLists.txt (hunk 8 above)
Comment (kunichik) [reply src/ipc/local/CMakeLists.txt]: done

File: src/ipc/ipc-common/LocalProviderBinderBase.cpp (hunk 4 above)
Comment (kunichik) [reply src/ipc/ipc-common/LocalProviderBinderBase.cpp]: done

File: src/ipc/ipc-common/IPCProxyBaseBase.h (hunk 3 above)
Comment (kunichik) [reply src/ipc/ipc-common/IPCProxyBaseBase.h]: done

File: src/ipc/dbus/DBusIPCProxyBinder.h (hunk 2 above)
Comment (kunichik) [reply src/ipc/dbus/DBusIPCProxyBinder.h]: done

File: src/ipc/dbus/DBusIPCProxyBinder.h (hunk 1 above)
Comment (kunichik) [reply src/ipc/dbus/DBusIPCProxyBinder.h]: done

File: src/ipc/local/LocalIPCMessage.cpp [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#include "LocalIPCMessage.h"
+
+#include <QObject>
+#include <QTextStream>
+
+#include "LocalIPC-serialization.h"
```
Comment (bitmouse) [inline src/ipc/local/LocalIPCMessage.cpp:36]: organize includes better, e.g. `#include <...> ` on top, `#include "..."` below

File: src/model/InterfaceBase.h [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+**
+**********************************************************************/
+
+#pragma once
+
+//#include <memory>
+//#include <functional>
+
+#include <QObject>
+#include <QMap>
```
Comment (bitmouse) [inline src/model/InterfaceBase.h:37]: keep includes consistent (see one of previous comments)

File: src/model/StructureFactoryBase.h [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+
+#include <QObject>
+#include <QQmlEngine>
+
+#include "StringConversionHandler.h"
```
Comment (bitmouse) [inline src/model/StructureFactoryBase.h:36]: do we need this include?

File: src/model/TAsyncAnswerMaster.h [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+
+//#include <memory>
+//#include <functional>
```
Comment (bitmouse) [inline src/model/TAsyncAnswerMaster.h:34]: remove

File: src/model/TProperty.h [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+        triggerValueChangedSignal();
+    }
+
+    Type m_previousValue = m_value;  /// The value when the last "value changed" signal was triggered
+
+    GetterFunction m_getterFunction;  /// The bound getter function, if any
+
+};
+
+}
```
Comment (bitmouse) [inline src/model/TProperty.h:209]: add new line at the end of file

File: src/model/TQMLAdapter.h [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+        registerInterfaceImplementationInstance(m_provider);
+    }
+
+private:
+    ProviderImplementationType m_provider;
+
+};
+
+
+}
```
Comment (bitmouse) [inline src/model/TQMLAdapter.h:61]: new line

File: src/model/TQMLImplListProperty.h [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+#include <QObject>
+#include <QList>
+#include <QVariant>
+
+#include "TProperty.h"
+#include "QMLImplListPropertyBase.h"
+//#include "FaceliftConversion.h"
+//#include "FaceliftQMLUtils.h"
+
+//#include "StructQObjectWrapperBase.h"
```
Comment (bitmouse) [inline src/model/TQMLImplListProperty.h:43]: remove

File: src/model/TQMLImplMapProperty.h [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+private:
+    Property<QMap<QString, ElementType> > *m_property = nullptr;
+    QVariantMap m_assignedVariantMap;
+    QList<QMetaObject::Connection> m_changeSignalConnections;
+};
+
+
+}
```
Comment (bitmouse) [inline src/model/TQMLImplMapProperty.h:133]: new line

File: src/ipc/ipc-common/CMakeLists.txt [hunk 17]
```diff
@@ -2,24 +2,32 @@
 facelift_add_library(FaceliftIPCCommonLib
     SOURCES
         ipc-common.cpp
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/ipc-common/CMakeLists.txt:4]: yep we use the Logging category "facelift.ipc"

File: src/ipc/ipc-common/CMakeLists.txt [hunk 18]
```diff
@@ -2,24 +2,32 @@
...
-        IPCProxyBase.cpp
+        IPCProxyBaseBase.cpp
         ModuleIPCBase.cpp
         ipc-serialization.cpp
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/ipc-common/CMakeLists.txt:7]: yes, we use it in the template facelift/codegen/facelift/templates/IPCProxyAdapter.template.cpp

File: src/ipc/ipc-common/CMakeLists.txt [hunk 19]
```diff
@@ -2,24 +2,32 @@
...
+        IPCAttachedPropertyFactory.cpp
         IPCProxyBinderBase.cpp
         InterfaceManager.cpp
         IPCProxyNewBase.cpp
         NewIPCServiceAdapterBase.cpp
+        OutputPayLoad.cpp
+        InputPayLoad.cpp
+        LocalProviderBinderBase.cpp
     HEADERS
         ipc-common.h
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/ipc-common/CMakeLists.txt:20]: we use  OutputPayLoad/InputPayLoad in the serialization/deserialization process respectively

File: src/ipc/ipc-common/CMakeLists.txt [hunk 20]
```diff
@@ -29,8 +37,15 @@ facelift_add_library(FaceliftIPCCommonLib
         ModuleIPCBase.h
         ipc-serialization.h
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/ipc-common/CMakeLists.txt:38]: we include ipc-serialization.h for using operator<<()  and others operators for manipulating with OutputPayLoad and InputPayLoad.
We use it in 
facelift/src/common/FaceliftUtils.h:86:15
the analysis is this code actually useful is out of scope of this task.

File: src/ipc/local/LocalIPCProxy.h [hunk 21]
```diff
@@ -45,108 +46,7 @@ namespace local {
...
-    {
-    }
-
-protected:
-    LocalIPCProxyBinder &m_ipcBinder;
-    bool m_serviceRegistered = false;
-
-};
-
+//class LocalIPCServiceAdapterBase;
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/local/LocalIPCProxy.h:49]: done

File: src/model/TAsyncAnswerMaster.h (hunk 12 above)
Comment (kunichik) [reply src/model/TAsyncAnswerMaster.h:34]: done

File: src/model/TProperty.h (hunk 13 above)
Comment (kunichik) [reply src/model/TProperty.h:209]: done

File: src/model/TQMLAdapter.h (hunk 14 above)
Comment (kunichik) [reply src/model/TQMLAdapter.h:61]: done

File: src/model/TQMLImplListProperty.h (hunk 15 above)
Comment (kunichik) [reply src/model/TQMLImplListProperty.h:43]: done

File: src/model/TQMLImplMapProperty.h (hunk 16 above)
Comment (kunichik) [reply src/model/TQMLImplMapProperty.h:133]: done

File: src/ipc/local/LocalIPCProxyBase.h [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+#  define FaceliftIPCLocalLib_EXPORT Q_DECL_EXPORT
+#else
+#  define FaceliftIPCLocalLib_EXPORT Q_DECL_IMPORT
+#endif
+
+namespace facelift {
+
+namespace local {
+
+//using namespace facelift;
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/local/LocalIPCProxyBase.h:48]: done

File: src/ipc/local/LocalIPCServiceAdapterBase.cpp [hunk 23]
```diff
@@ -0,0 +1,160 @@
...
+
+
+// #include "FaceliftModel.h"
+// #include "FaceliftUtils.h"
+// #include "ModelProperty.h"
+
+// #include "QMLAdapter.h"
+// #include "QMLModel.h"
+
+//#include "LocalIPCProxy.h"
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/local/LocalIPCServiceAdapterBase.cpp:43]: done

File: src/model/.directory [hunk 24]
```diff
@@ -0,0 +1,4 @@
+[Dolphin]
+Timestamp=2020,7,28,12,17,18
+Version=4
+ViewMode=2
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/.directory:4]: done, sorry :)

File: src/model/InterfaceBase.h [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+
+//#include <memory>
+//#include <functional>
+
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/InterfaceBase.h:35]: done

File: src/model/QMLAdapterBase.h [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+#include <QQmlEngine>
+#include <QtQml>
+#include <QQmlParserStatus>
+
+#include "FaceliftModel.h"
+
+namespace facelift {
+
+//class IPCAttachedPropertyFactory;
+//class IPCServiceAdapterBase;
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/QMLAdapterBase.h:43]: done

File: src/model/QMLModelImplementationFrontendBase.h [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+** CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+
+#include <QQmlEngine>
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/QMLModelImplementationFrontendBase.h:33]: done, there is enough  the forward declaration

File: src/model/ServicePropertyInterface.h [hunk 28]
```diff
@@ -0,0 +1,83 @@
+
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/ServicePropertyInterface.h:1]: done

File: src/model/ServiceRegistry.h [hunk 29]
```diff
@@ -58,6 +58,7 @@ class FaceliftModelLib_EXPORT ServiceRegistry : public QObject
         return m_objects;
     }
 
+
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/ServiceRegistry.h:61]: done

File: src/model/ServiceWrapperBase.h [hunk 30]
```diff
@@ -0,0 +1,84 @@
...
+        return ((newService->*getter)() != referenceValue);
+    }
+
+private:
+    QVector<QMetaObject::Connection> m_connections;  /// The list of connections which this property is bound to
+
+};
+
+
+}
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/model/ServiceWrapperBase.h:84]: done

File: src/ipc/ipc-common/IPCProxyBase.h [hunk 31]
```diff
@@ -114,5 +108,4 @@ class IPCProxyBase : public AdapterType, protected IPCProxyBaseBase
 
 };
 
-
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/ipc-common/IPCProxyBase.h:117]: OK, i will.

File: src/ipc/local/LocalIPCMessage.cpp (hunk 9 above)
Comment (kunichik) [reply src/ipc/local/LocalIPCMessage.cpp:36]: Ok, done

File: src/model/StructureFactoryBase.h (hunk 11 above)
Comment (kunichik) [reply src/model/StructureFactoryBase.h:36]: definitely not, deleted, but some other changes are present

File: src/ipc/local/LocalIPCServiceAdapterBase.cpp [hunk 32]
```diff
@@ -0,0 +1,151 @@
...
+** CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#include "LocalIPCMessage.h"
+
+#include <QObject>
```
Comment (bitmouse) [inline src/ipc/local/LocalIPCServiceAdapterBase.cpp:33]: swap these includes if possible

File: src/model/QMLModelImplementationFrontendBase.h [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+namespace facelift {
+
+class qmlEngine;
```
Comment (bitmouse) [inline src/model/QMLModelImplementationFrontendBase.h:34]: does it even compile? I think this should be QQmlEngine. Please check.

File: src/ipc/local/LocalIPCServiceAdapterBase.cpp (hunk 32 above)
Comment (kunichik) [reply src/ipc/local/LocalIPCServiceAdapterBase.cpp:33]: done

File: src/ipc/ipc-common/ipc-serialization.h [hunk 34]
```diff
@@ -36,227 +36,19 @@
...
-            ElementType value;
-            IPCTypeHandler<QString>::read(msg, key);
-            IPCTypeHandler<ElementType>::read(msg, value);
-            map.insert(key, value);
-        }
-    }
-};
-
+/*
 template<typename Type>
```
[the remark this answers is not in the document]
Comment (kunichik) [reply src/ipc/ipc-common/ipc-serialization.h:52]: it was a mistake, already fixed in previous version

File: src/model/QMLModelImplementationFrontendBase.h (hunk 33 above)
Comment (kunichik) [reply src/model/QMLModelImplementationFrontendBase.h:34]: OMG - it is in the facelift namespace in addition :)
done

**Comment by kunichik** [comment]
> Please check why it does not compile.

done, but I cannot resolve this - I'm going to dismiss

**Pull request merged**